heron/executor/src/python/heron_executor.py

```diff
@@ -200,8 +248,19 @@ def initialize(self):
     constructor
     """
     create_folders = 'mkdir -p %s' % self.log_dir
-    chmod_binaries = 'chmod a+rx . && chmod a+x %s && chmod +x %s %s %s' \
-        % (self.log_dir, self.tmaster_binary, self.stmgr_binary, self.heron_shell_binary)
+    chmod_x_binaries = [self.tmaster_binary, self.stmgr_binary, self.heron_shell_binary]
+    binaries_to_chmod = []
+
+    for binary in chmod_x_binaries:
+      stat_result = os.stat(binary)[stat.ST_MODE]
+      if not stat_result & stat.S_IXOTH:
+        binaries_to_chmod.append(binary)
```

Comment:
Works for me, I'll make that change. Stat appears to work on everything major, but I'll do some additional digging and testing on some VMs to be sure.